zerver/lib/bugdown/__init__.py

```diff
@@ -216,7 +217,21 @@ def add_a(
     if data_id is not None:
         a.set("data-id", data_id)
     img = markdown.util.etree.SubElement(a, "img")
-    img.set("src", url)
+    if use_thumbnails:
+        thumbnail_path = '/thumbnail'
+        if not url.startswith('/'):
+            thumbnail_path += '/'
+        img.set("src", "{0}{1}?size=0x100".format(
+            thumbnail_path,
+            urllib.parse.quote(url)
+        ))
+        img.set('data-original', "{0}{1}?size=0x0".format(
+            thumbnail_path,
+            urllib.parse.quote(url)
+        ))
+    else:
+        img.set("src", url)
```

Comment:
Are there any call points that are not using `use_thumbnails`?  I kinda wonder if we want to flip this around, so the default is True and we only specify in the cases where we don't want a thumbnail...